File: docs/current/guides/snippets/use-services/test-service-lifecycle-2/main.py
import sys

import anyio
import dagger


async def main():
    # create Dagger client
    async with dagger.Connection(dagger.Config(log_output=sys.stderr)) as client:
        # create Redis service container
        redisSrv = (
            client.container().from_("redis").with_exposed_port(6379).with_exec([])
        )

        # create Redis client container
        redisCli = (
            client.container()
            .from_("redis")
            .with_service_binding("redis-srv", redisSrv)
            .with_entrypoint(["redis-cli", "-h", "redis-srv"])
        )

        # set value
        setter = await redisCli.with_exec(["set", "foo", "abc"]).stdout()

        # get value
        getter = await redisCli.with_exec(["get", "foo"]).stdout()

    print(setter)
    print(getter)


anyio.run(main)

File: docs/current/guides/snippets/use-services/test-service-lifecycle-2/main.go
package main

import (
	"context"
	"fmt"
	"os"

	"dagger.io/dagger"
)

func main() {
	ctx := context.Background()

	// create Dagger client
	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))

	if err != nil {
		panic(err)
	}
	defer client.Close()

	// create Redis service container
	redisSrv := client.Container().
		From("redis").
		WithExposedPort(6379).
		WithExec(nil)

	// create Redis client container
	redisCLI := client.Container().
		From("redis").
		WithServiceBinding("redis-srv", redisSrv).
		WithEntrypoint([]string{"redis-cli", "-h", "redis-srv"})

	// set value
	setter, err1 := redisCLI.
		WithExec([]string{"set", "foo", "abc"}).
		Stdout(ctx)

	if err1 != nil {
		panic(err1)
	}

	fmt.Println(setter)

	// get value
	getter, err2 := redisCLI.
		WithExec([]string{"get", "foo"}).
		Stdout(ctx)

	if err2 != nil {
		panic(err2)
	}

	fmt.Println(getter)
}
